File: example/sdk-integration/src/main/java/com/fawry/softpos/android/example/sdk/FawryOperation.kt
package com.fawry.softpos.android.example.sdk

/**
 * Operations exposed by the demo apps.
 *
 * Each enum maps one UI option to the matching Fawry SDK request builder in [FawrySdkClient].
 */
enum class FawryOperation(val label: String, val actionLabel: String, val needsReference: Boolean) {
    CARD_SALE("Card Sale", "Pay Now", false),
    CARD_REFUND("Card Refund", "Refund", true),
    CARD_VOID("Card Void", "Void", true),
    INQUIRY("Inquiry", "Run Inquiry", true),
    CLEAR_CACHE("Clear Cache", "Clear Cache", false);

    companion object {
        val labels = entries.map { it.label }
    }
}

File: tapnpay_flutter_example/lib/models/fawry_operation.dart
/// Operations supported by this demo, mirroring `FawryOperation` in
/// `example/sdk-integration` on the native Android side.
enum FawryOperation {
  cardSale('Card Sale', 'Pay Now', false),
  cardRefund('Card Refund', 'Refund', true),
  cardVoid('Card Void', 'Void', true),
  inquiry('Inquiry', 'Run Inquiry', true),
  clearCache('Clear Cache', 'Clear Cache', false);

  const FawryOperation(this.label, this.actionLabel, this.needsReference);

  final String label;
  final String actionLabel;
  final bool needsReference;

  /// Matches the Kotlin enum constant name (e.g. `CARD_SALE`) expected by
  /// `TapNPayBridge` on the native side.
  String get nativeName => switch (this) {
        FawryOperation.cardSale => 'CARD_SALE',
        FawryOperation.cardRefund => 'CARD_REFUND',
        FawryOperation.cardVoid => 'CARD_VOID',
        FawryOperation.inquiry => 'INQUIRY',
        FawryOperation.clearCache => 'CLEAR_CACHE',
      };
}
